File: games/sandbox-v1/src/server/MockPlayer.lua
-- ServerScriptService/MockPlayer.lua
local MockPlayer = {}
MockPlayer.__index = MockPlayer

function MockPlayer.new(name, userId, participantType)
    local self = setmetatable({}, MockPlayer)
    self.Name = name
    self.DisplayName = name
    self.UserId = userId or -1  -- Keep negative ID for backwards compatibility
    self.Type = participantType or "npc"  -- Default to "npc" if not specified
    return self
end

function MockPlayer:IsA(className)
    return className == "Player"
end

-- Add helper method to check participant type
function MockPlayer:GetParticipantType()
    return self.Type
end

return MockPlayer

File: api/app/mock_player.py
class MockPlayer:
    """Mock player class for NPC-NPC interactions"""
    def __init__(self, display_name: str, npc_id: str, participant_type: str = "npc"):
        self.display_name = display_name
        self.npc_id = npc_id
        self.participant_type = participant_type
        self.Name = display_name  # For compatibility with Player interface
        
    def is_player(self) -> bool:
        return False
        
    def __str__(self) -> str:
        return f"MockPlayer({self.display_name}, {self.npc_id})" 